### PR TITLE
Standardize notebook pip installs

### PR DESCRIPTION
**Pull request by balopat**
Makes pip installs standard in notebooks. 

pip installs were all over in our notebooks, sometimes missing, sometimes pinned to a version, sometimes installed pip as well, and never really checked whether it is important to install cirq at all. In case of running jupyter notebook locally from the repo, there is no need to run pip install. In case the notebook is running in colab, we always have to run pip install (at least the first time in a runtime). 

Also added the following fixes in Google notebooks: 
- project_id is now empty by default and throws an error if it's empty like this: 

image

You can try these out in Colab directly from my fork:
[cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb)
[cirq/contrib/quimb/Contract-a-Grid-Circuit.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/cirq/contrib/quimb/Contract-a-Grid-Circuit.ipynb)
[cirq/contrib/svg/example.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/cirq/contrib/svg/example.ipynb)
[docs/circuits.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/circuits.ipynb)
[docs/gates.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/gates.ipynb)
[docs/google/concepts.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/google/concepts.ipynb)
[docs/interop.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/interop.ipynb)
[docs/noise.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/noise.ipynb)
[docs/pasqal/getting_started.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/pasqal/getting_started.ipynb)
[docs/qudits.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/qudits.ipynb)
[docs/simulation.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/simulation.ipynb)
[docs/tutorials/aqt/getting_started.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/aqt/getting_started.ipynb)
[docs/tutorials/basics.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/basics.ipynb)
[docs/tutorials/google/colab.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/google/colab.ipynb)
[docs/tutorials/google/reservations.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/google/reservations.ipynb)
[docs/tutorials/google/start.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/google/start.ipynb)
[docs/tutorials/hidden_linear_function.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/hidden_linear_function.ipynb)
[docs/tutorials/qaoa.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/qaoa.ipynb)
[docs/tutorials/quantum_walks.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/quantum_walks.ipynb)
[docs/tutorials/rabi_oscillations.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/rabi_oscillations.ipynb)
[docs/tutorials/variational_algorithm.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/docs/tutorials/variational_algorithm.ipynb)
[examples/advanced/quantum_volume.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/examples/advanced/quantum_volume.ipynb)
[examples/advanced/quantum_volume_errors.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/examples/advanced/quantum_volume_errors.ipynb)
[examples/advanced/quantum_volume_routing.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/examples/advanced/quantum_volume_routing.ipynb)
[examples/direct_fidelity_estimation.ipynb](https://colab.sandbox.google.com/github/balopat/Cirq/blob/fix_notebook_pip_installs/examples/direct_fidelity_estimation.ipynb)

### CODE REVIEW

**Comment by mpharrigan** [comment]
Great minds think alike! I just opened https://github.com/quantumlib/ReCirq/issues/83 for ReCirq.

Did you consider doing try/except for import error? Avoids the extra import of importlib and is closer to Python's "ask for forgiveness not permission" zen mantra

File: cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import importlib.util\n",
+    "\n",
+    "if importlib.util.find_spec('cirq') is None: \n",
+    "    print(\"installing cirq\")\n",
+    "    !pip install --quiet cirq\n",
```
Comment (viathor) [inline cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb:13]: Optional: Running the shell command comes with a slight risk that pip will install the package into a location different from the one used by the colab kernel. There is a special colab magic `%pip` that ensures installation into the appropriate location. Not sure when it was added though.

**Review by viathor — APPROVED** [review]

**Comment by balopat** [comment]
> Great minds think alike! I just opened [quantumlib/ReCirq#83](https://github.com/quantumlib/ReCirq/issues/83) for ReCirq.
> 
> Did you consider doing try/except for import error? Avoids the extra import of importlib and is closer to Python's "ask for forgiveness not permission" zen mantra

Great, I was thinking that we could standardize across the board. I started with the try / except form but then I backed out for two - not that great - reasons:
1.) I wanted to avoid double `cirq import` (which could be avoided by removing them from the other places in the notebook first place, but even if they are there it is not the end of the world due to import caching) 
2.) I think I had a "don't use exceptions for flow control" mantra in my head from my java days :D - which in python are [much less of an issue](https://softwareengineering.stackexchange.com/questions/351110/are-exceptions-for-flow-control-best-practice-in-python)

Thanks for chiming in - I'll change it to exception based form instead.

File: cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb (hunk 1 above)
Comment (balopat) [reply cirq/contrib/quimb/Cirq-to-Tensor-Networks.ipynb:13]: Huh, let me look into this, thanks @viathor!

**Comment by mpharrigan** [comment]
Oh, I didn't think of (1). From a performance perspective, it shouldn't matter because of the caching you identify but I acknowledge that it's not aesthetically pleasing. Maybe it can be put in the except clause?

```python
try:
  import cirq
except ImportError
  !pip ...
  import cirq
```

---

Has it ever been discussed whether we can get Cirq pre-installed into a colab runtime? This might be the ultimate cleanest solution

**Comment by balopat** [comment]
> Oh, I didn't think of (1). From a performance perspective, it shouldn't matter because of the caching you identify but I acknowledge that it's not aesthetically pleasing. Maybe it can be put in the except clause?
> 
> ```python
> try:
>   import cirq
> except ImportError
>   !pip ...
>   import cirq
> ```

That can work. I'm fine with this. It's less of an issue than the extra `importlib` import in the namespace.
 
> Has it ever been discussed whether we can get Cirq pre-installed into a colab runtime? This might be the ultimate cleanest solution

It's an idea worth exploring, and I've also been thinking about it. I did have protobuf updated in colab in July - it is a lengthy (~1-2 weeks) process but not the end of the world. We'd have to do that for every release though. It would definitely speed up things for users, with a bit of overhead from us. I'm open exploring it but we'd have to commit to having a stable process around it - I'd hate to have outdated cirq in colab runtimes.

However, regardless whether colab has cirq - I think it is still useful to have the installation script in notebooks for other circumstances to make them portable, e.g. jupyter-notebook locally, without a cirq installation or using binder or other platforms that I don't know about.

**Pull request merged**